File: sdks/python/examples/events/filter.py
from examples.events.worker import EVENT_KEY, event_workflow
from hatchet_sdk import Hatchet, PushEventOptions

hatchet = Hatchet()

# > Create a filter
for i in range(50):
    hatchet.filters.create(
        workflow_id=event_workflow.id,
        expression="input.should_skip == false",
        scope="foobarbaz",
        payload={
            "main_character": "Anna",
            "supporting_character": "Stiva",
            "location": "Moscow",
            "index": str(i),
        },
    )
# !!

# > Skip a run
hatchet.event.push(
    event_key=EVENT_KEY,
    payload={
        "should_skip": True,
    },
    options=PushEventOptions(
        scope="foobarbaz",
    ),
)
# !!

# > Trigger a run
hatchet.event.push(
    event_key=EVENT_KEY,
    payload={
        "should_skip": False,
    },
    options=PushEventOptions(
        scope="foobarbaz",
    ),
)
# !!

File: sdks/ruby/examples/events/filter.rb
# frozen_string_literal: true

require "hatchet-sdk"
require_relative "worker"

HATCHET_CLIENT = Hatchet::Client.new

# > Create a filter
HATCHET_CLIENT.filters.create(
  workflow_id: EVENT_WORKFLOW.id,
  expression: "input.should_skip == false",
  scope: "foobarbaz",
  payload: {
    "main_character" => "Anna",
    "supporting_character" => "Stiva",
    "location" => "Moscow"
  }
)
# !!

# > Skip a run
HATCHET_CLIENT.event.push(
  EVENT_KEY,
  { "should_skip" => true },
  scope: "foobarbaz"
)
# !!

# > Trigger a run
HATCHET_CLIENT.event.push(
  EVENT_KEY,
  { "should_skip" => false },
  scope: "foobarbaz"
)
# !!
